custom_components/ha_washdata/diag_buffer.py

```
@@ -0,0 +1,184 @@
+        # Match the exact prefix added by DeviceLoggerAdapter
+        self._tag = f"[{device_name}]"
+        # Store (created_float, levelname, message)
+        self._buf: deque[tuple[float, str, str]] = deque(maxlen=_MAX_LOGS)
+
+    def emit(self, record: logging.LogRecord) -> None:
+        try:
+            msg = record.getMessage()
+            if self._tag not in msg:
+                return
+            self._buf.append((record.created, record.levelname, msg))
+        except Exception:  # pylint: disable=broad-except
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)